### DISCUSSION TITLE
Publish macOS arm64 build with GitHub releases

### DISCUSSION THREAD

**Discussion by Nezteb** [odin-lang/Odin#2830] [Proposals]
Currently [Odin publishes builds](https://github.com/odin-lang/Odin/releases) for the following platforms:

- `macos-amd64`
- `ubuntu-amd64`
- `windows-amd64`

I'd also like to see a version for `macos-arm64` so that macOS users with newer hardware don't have to compile Odin from source if they would prefer.

I assume the absence of arm64 builds is because GitHub Actions don't support hosted macOS arm64 runners ([yet?](https://github.com/actions/runner/issues/805#issuecomment-1438149537)). [They do support self-hosted runners](https://github.blog/changelog/2022-08-09-github-actions-self-hosted-runners-now-support-apple-m1-hardware/), but that would require someone on the core team to have a dedicated M1/M2 machine running it.

A simpler alternative would be to have someone with an M1/M2 manually build Odin and add it to the GitHub release whenever it is ready.

What do y'all think?

EDIT: ["Apple Silicon (M1) powered macOS runners are now available in public beta"](https://github.com/actions/runner-images/issues/8439)

**Comment by laytan** [comment]
I made a PR to add this here: https://github.com/odin-lang/Odin/pull/3163

**Comment by pathscale** [comment]
While the GH runners are available for enterprise.. I don't know if it works for an open source project like this.

However.. buildjet is awesome, cheap and super easy to setup.

I'd be happy to donate the $10 required (if needed)

**Reply by Kelimion** [reply]
Github added mac/arm64 support a while ago, and Laytan's PR was merged. Closing out this discussion. Thanks for the offer, though.
